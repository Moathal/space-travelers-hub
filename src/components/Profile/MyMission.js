import React from 'react';

function MyMission({ name }) {
  return (
    <h4>{name}</h4>
  );
}

export default MyMission;
