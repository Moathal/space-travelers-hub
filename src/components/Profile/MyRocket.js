import React from 'react';

function MyRocket({ name }) {
  return (
    <h4>{name}</h4>
  );
}

export default MyRocket;
