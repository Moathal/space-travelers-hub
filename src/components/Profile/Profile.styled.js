import styles from 'styled-components';

const Cards = styles.div`
    width: 100%;
    display: flex;
    align-items: flex start;
    justify-content: space-evenly;
    margin-top: 2rem;
`;

export default Cards;
