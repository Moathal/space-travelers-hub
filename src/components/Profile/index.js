import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.styles.scss';

const Profile = () => {
  const missions = useSelector(
    (state) => state.missions.missions.filter((mission) => mission.reserved === true),
  );

  return (
    <div className="profile-container">
      <div className="missions-container">
        <div className="title">My Missions</div>
        <ul className="my-missions">
          {
            missions && missions.map((mission) => (
              <li key={mission.id} className="my-missions-item">{mission.mission_name}</li>
            ))
          }
        </ul>
      </div>
      <div className="rockets-container">
        <div className="title">My Rockets</div>
        <ul className="my-rockets">
          <li className="my-rockets-item">Rocket 1</li>
          <li className="my-rockets-item">Rocket 2</li>
          <li className="my-rockets-item">Rocket 3</li>
          <li className="my-rockets-item">Rocket 4</li>
        </ul>
      </div>
    </div>
  );
};

export default Profile;
