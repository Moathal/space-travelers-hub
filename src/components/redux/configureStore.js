import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions';
import rocketsReducer from './rockets';

export default configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice,
  },
});
