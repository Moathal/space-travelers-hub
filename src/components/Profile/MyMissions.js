import React from 'react';
import { useSelector } from 'react-redux';
import MyMission from './MyMission';

function MyMissions() {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div>
      <h2>My Missions</h2>
      {joinedMissions.map((mission) => (
        <MyMission key={mission.id} name={mission.mission_name} />
      ))}
    </div>
  );
}

export default MyMissions;
