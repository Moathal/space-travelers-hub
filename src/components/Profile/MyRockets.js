import React from 'react';
import { useSelector } from 'react-redux';
import MyRocket from './MyRocket';

function MyRockets() {
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div>
      <h2>My Rockets</h2>
      { reservedRockets.map((rocket) => (
        <MyRocket key={rocket.id} name={rocket.rocket_name} />
      ))}
    </div>
  );
}

export default MyRockets;
