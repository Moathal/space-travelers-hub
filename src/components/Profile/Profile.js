import React from 'react';
import MyMissions from './MyMissions';
import MyRockets from './MyRockets';
import Cards from './Profile.styled';

function Profile() {
  return (
    <Cards>
      <MyMissions />
      <MyRockets />
    </Cards>
  );
}

export default Profile;
