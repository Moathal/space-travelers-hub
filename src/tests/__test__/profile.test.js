import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../components/redux/configureStore';
import Profile from '../../components/Profile/Profile';

it('renders Profile correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
