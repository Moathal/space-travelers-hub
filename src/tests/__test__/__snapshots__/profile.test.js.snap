// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile correctly 1`] = `
<div
  className="sc-bcXHqe hKeyby"
>
  <div>
    <h2>
      My Missions
    </h2>
  </div>
  <div>
    <h2>
      My Rockets
    </h2>
  </div>
</div>
`;
